src/tools/r2rdump/R2RMethod.cs

```diff
@@ -130,21 +107,170 @@ class R2RMethod
+            }
+
+            ReturnType = new SignatureType(ref signatureReader, mdReader, genericParams);
+            ArgTypes = new SignatureType[argCount];
+            for (int i = 0; i < argCount; i++)
+            {
+                ArgTypes[i] = new SignatureType(ref signatureReader, mdReader, genericParams);
+            }
+
             Token = _mdtMethodDef | rid;
+        }
+
+        /// <summary>
+        /// Set the entry point id for generic methods and maps the generic parameters to the type
+        /// </summary>
+        public R2RMethod(byte[] image, MetadataReader mdReader, uint rid, int entryPointId, GenericElementTypes[] instanceArgs, uint[] tok)
+            : this(image, mdReader, rid)
+        {
+            EntryPointRuntimeFunctionId = entryPointId;
+
+            Dictionary<string, string> tokens = new Dictionary<string, string>();
+            for (int i = 0; i < _genericParamInstance.Count; i++)
+            {
+                var key = _genericParamInstance.ElementAt(i).Key;
+                _genericParamInstance[key] = instanceArgs[i];
+
+                if (instanceArgs[i] == GenericElementTypes.ValueType) {
+                    tokens[key] = mdReader.GetString(mdReader.GetTypeDefinition(MetadataTokens.TypeDefinitionHandle((int)tok[i])).Name);
+                }
+            }
+
+            if ((ReturnType.Flags & SignatureType.SignatureTypeFlags.GENERIC) != 0)
+            {
+                GenericElementTypes instance = _genericParamInstance[ReturnType.TypeName];
+                if (instance == GenericElementTypes.ValueType)
+                {
+                    ReturnType.GenericInstance = new GenericInstance(instance, tokens[ReturnType.TypeName]);
+                }
+                else
+                {
+                    ReturnType.GenericInstance = new GenericInstance(instance, Enum.GetName(typeof(GenericElementTypes), instance));
+                }
+            }
+
+            for (int i = 0; i<ArgTypes.Length; i++)
+            {
+                if ((ArgTypes[i].Flags & SignatureType.SignatureTypeFlags.GENERIC) != 0)
+                {
+                    GenericElementTypes instance = _genericParamInstance[ArgTypes[i].TypeName];
+                    if (instance == GenericElementTypes.ValueType)
+                    {
+                        ArgTypes[i].GenericInstance = new GenericInstance(instance, tokens[ArgTypes[i].TypeName]);
+                    }
+                    else
+                    {
+                        ArgTypes[i].GenericInstance = new GenericInstance(instance, Enum.GetName(typeof(GenericElementTypes), instance));
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Uses the methodEntryPoint native array to get the entry point id of non-generic methods
+        /// </summary>
+        public R2RMethod(byte[] image, MetadataReader mdReader, NativeArray methodEntryPoints, uint rid, int offset)
```

Comment:
It looks like the methodEntryPoints argument is no longer needed.

Reply:
Is it possible to use the offset to calculate the EntryPointRuntimeFunctionId outside of this constructor and just pass it in as a parameter like the previous constructor?

Reply:
What I'd like to achieve here is to remove this constructor. If both the main constructors can receive the entrypointId as an argument, then that can be added to the constructor with the fewest parameters.